app/src/androidTest/java/com/hedvig/app/feature/embark/EmbarkMenuTest.kt

```diff
@@ -0,0 +1,129 @@
+package com.hedvig.app.feature.embark
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.agoda.kakao.intent.KIntent
+import com.agoda.kakao.screen.Screen
+import com.hedvig.android.owldroid.graphql.EmbarkStoryQuery
+import com.hedvig.app.feature.embark.screens.EmbarkScreen
+import com.hedvig.app.feature.embark.screens.TextActionScreen
+import com.hedvig.app.feature.embark.ui.EmbarkActivity
+import com.hedvig.app.feature.embark.ui.MoreOptionsActivity
+import com.hedvig.app.feature.settings.SettingsActivity
+import com.hedvig.app.feature.zignsec.ZignSecAuthenticationActivity
+import com.hedvig.app.testdata.feature.embark.data.STORY_WITH_TEXT_ACTION
+import com.hedvig.app.util.ApolloCacheClearRule
+import com.hedvig.app.util.ApolloMockServerRule
+import com.hedvig.app.util.LazyIntentsActivityScenarioRule
+import com.hedvig.app.util.apolloResponse
+import com.hedvig.app.util.context
+import com.hedvig.app.util.stub
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import org.junit.Rule
+import org.junit.Test
+
+class EmbarkMenuTest : TestCase() {
+
+    @get:Rule
+    val activityRule = LazyIntentsActivityScenarioRule(EmbarkActivity::class.java)
+
+    @get:Rule
+    val apolloMockServerRule = ApolloMockServerRule(
+        EmbarkStoryQuery.QUERY_DOCUMENT to apolloResponse { success(STORY_WITH_TEXT_ACTION) }
+    )
+
+    @get:Rule
+    val apolloCacheClearRule = ApolloCacheClearRule()
+
+    val authIntent = KIntent {
+        hasComponent(ZignSecAuthenticationActivity::class.java.name)
+    }
+
+    val appInfoIntent = KIntent {
+        hasComponent(MoreOptionsActivity::class.java.name)
+    }
+
+    val settingsIntent = KIntent {
+        hasComponent(SettingsActivity::class.java.name)
+    }
+
+    @Test
+    fun loginButtonShouldOpenLoginMethod() = run {
+        val intent = EmbarkActivity.newInstance(context(), this.javaClass.name)
+        activityRule.launch(intent)
+
+        TextActionScreen {
+            authIntent { stub() }
+
+            openActionBarOverflowOrOptionsMenu(ApplicationProvider.getApplicationContext())
+
+            onView(withText("Log in"))
```

Comment:
Can we express this using the `Kakao`-DSL? 🙂

Reply:
Tried but couldn't find a way 🤔

Reply:
```kotlin
val login = KView { withText("Log in") }
```

Should do the trick I think 😉

Reply:
Oh look at that, thought I tried that before 😅 cheers